templates/template.go

```diff
@@ -8,32 +8,77 @@ import (
 	"io"
 	"text/template"
 
+	"github.com/aws/aws-sdk-go/aws"
 	"github.com/gobuffalo/packr"
 )
 
-// NewTemplate will create a temp file with the template for a CFN stack
-func NewTemplate(assetName string, data interface{}) (io.Reader, error) {
+// GetAsset reads an asset from "disk" into a string
+func GetAsset(assetName string, options ...func(string, *string) error) (string, error) {
```

Comment:
It can! I actually turned all assetNames into consts in type.go but stopped short of giving them their own type just because of the massive number of places I found it referenced. I will go through though and update all method signatures accepting an assetName to use a new asset type. This would take care of the first argument.

The second argument is a pointer to the template. I'll go through and see where this would need to be updated in other methods so that we don't have to do any type coercion.

Reply:
Sorry, i wasn't referring to the first parameter...rather the second parameter of type `func(string, *string) error`.  I was suggesting a new type like:

```
type AssetOption func(string, *string) error
```

Reply:
Great suggestion! Thank you - that's much better than what I had.